### DISCUSSION TITLE
Set release's associated branch

### DISCUSSION THREAD

**Discussion by dgw** [community/community#21761] [New to GitHub]
<p>I tagged a couple pre-releases of a side project today, after noticing that they had not been tagged at the time.</p>
<p>Here they are: <a href="https://github.com/sopel-irc/sopel-github/releases/tag/0.2.0a1" rel="nofollow noopener">release 1</a>, <a href="https://github.com/sopel-irc/sopel-github/releases/tag/0.2.0a2" rel="nofollow noopener">release 2</a></p>
<p>They were created by using git tag  locally, then pushing those tags to GitHub and editing the tags there to add release notes.</p>
<p>With the understanding that the timestamps can be wonky when a release is created this way (I’ve spoken with support about that before), I’m confused that the “base” branch is considered to be master when the tagged commits do not exist on that branch. I’m releasing from an “alpha” branch that diverged quite some time ago.</p>
<p>Ultimately, it’s not <em>that</em> big a deal. I’m probably not even going to keep the pre-releases around anywhere (they’re only uploaded to PyPI so I can easily install test builds on remote systems), but it’s weird that the release UI shows a comparison to master in this case.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for reaching out.</p>
<p>Looking at your repository’s history, it seems that <a href="https://github.com/sopel-irc/sopel-github/releases/tag/0.2.0a0" rel="nofollow noopener">tag 0.2.0a0</a> shows “17 commits to alpha since release”, which is I suspect what you’re expecting from the other two releases that you linked to. However, <a href="https://github.com/sopel-irc/sopel-github/releases/tag/0.2.0a1" rel="nofollow noopener">tag 0.2.0a1</a> is the first to show “2 commits to master since release”. It appears that after tag 0.2.0a0, you began creating new branches  <strong>from <code>master</code></strong> to make changes and then merge them into alpha. Had you been making branches from alpha or using <code>git cherry-pick</code> to move the changes from master to alpha, I suspect you would have gotten the behavior you expect on the successive tags.</p>
<p>I hope that helps!</p>

**Comment by dgw** [comment]
<p>Yeah, it does. Thanks!</p>
<p>The merges from <code>master</code> are because I’m using <code>alpha</code> and its pre-releases to test upcoming changes “in production” on a dog-food instance before merging them into upcoming releases. When one of the PRs is updated, I merge again from that branch to pull in the newer changes.</p>
<p>There’s probably a better workflow to use the next time I need a pre-release branch, but for now I’ll just ignore this display issue.</p>
